test/test_testing.py

```diff
@@ -2828,20 +2828,28 @@ def __init__(self, framework: ops.Framework):
                 super().__init__(framework)
                 self.framework.observe(self.on.collect_app_status, self._on_collect_app_status)
                 self.framework.observe(self.on.collect_unit_status, self._on_collect_unit_status)
+                self.app_status_to_add = ops.BlockedStatus('blocked app')
+                self.unit_status_to_add = ops.BlockedStatus('blocked unit')
 
             def _on_collect_app_status(self, event: ops.CollectStatusEvent):
-                event.add_status(ops.ActiveStatus())
+                event.add_status(self.app_status_to_add)
 
             def _on_collect_unit_status(self, event: ops.CollectStatusEvent):
-                event.add_status(ops.BlockedStatus('bar'))
+                event.add_status(self.unit_status_to_add)
 
         harness = ops.testing.Harness(TestCharm)
         harness.set_leader(True)
         harness.begin()
         # Tests for the behaviour of status evaluation are in test_charm.py
         harness.evaluate_status()
-        self.assertEqual(harness.model.app.status, ops.ActiveStatus())
-        self.assertEqual(harness.model.unit.status, ops.BlockedStatus('bar'))
+        self.assertEqual(harness.model.app.status, ops.BlockedStatus('blocked app'))
+        self.assertEqual(harness.model.unit.status, ops.BlockedStatus('blocked unit'))
+
+        harness.charm.app_status_to_add = ops.ActiveStatus('active app')
+        harness.charm.unit_status_to_add = ops.ActiveStatus('active unit')
+        harness.evaluate_status()
+        self.assertEqual(harness.model.app.status, ops.ActiveStatus('active app'))
+        self.assertEqual(harness.model.unit.status, ops.ActiveStatus('active unit'))
```

Comment:
So evaluate_status calculates, commit somewhere else, and clears the collection for the next add_status?
Then should probably document that cannot do "additive tests" with evaluate_status.

Not sure which implementation is less confusing though.

Reply:
Good call. I've updated the `evaluate_status` docstring to clarify that. I don't think "additive tests" is useful for this: before triggering the collect-status events, the statuses added should be reset.